java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -372,15 +386,17 @@ private API getXPAPI(
       // if it's not a match, skip to next method property map
       String returnType = getValueStringFromMap(currentMethodProperties, RETURN_TYPE_STRING);
       if (returnType != null) {
-        if (!isMethodTreeMatchesMethodProperty(mst, RETURN_TYPE_STRING, returnType)) {
+        String mReturn = ASTHelpers.getReturnType(mst).toString();
+        if (!returnType.equalsIgnoreCase(mReturn)) {
```

Comment:
Why `equalsIgnoreCase` instead of case sensitive equals?

Reply:
Just to allow users the flexibility of wrongly capitalizing stuff. Like `java.lang.string` or the fully qualified name of their class.

Reply:
I say let's compare case-sensitive, since Java itself is case sensitive, I think, e.g. you can have both a type `com.foo.Bar` and one called `com.foo.bar` at the same time in the same JVM (you probably shouldn't, but still). That said, I have no super strong opinion here. cc: @mkr-plse Any thoughts?

Reply:
>...you can have both a type com.foo.Bar and one called com.foo.bar ...

Oh, didn't know this. Then a case sensitive match is a better default for now - just in case someone does this. 

Updated and fixed testcase.